### DISCUSSION TITLE
Why the systematic use of `builtin` or `command` in example bash scripts?

### DISCUSSION THREAD

**Discussion by grondilu** [kovidgoyal/kitty#6725] [General]
It seems to be a constant in all example scripts.

E.g., 
```bash
#!/bin/bash

# This uses the kitten standalone binary from kitty to get the pixel sizes
# since we can't do IOCTLs directly. Fortunately, kitten is a static exe
# pre-built for every Unix like OS under the sun.

builtin read -r rows cols < <(command stty size)
IFS=x builtin read -r width height < <(command kitten icat --print-window-size); builtin unset IFS
builtin echo "number of rows: $rows number of columns: $cols screen width: $width screen height: $height"
```

```bash
#!/bin/bash
transmit_png() {
    data=$(base64 "$1")
    data="${data//[[:space:]]}"
    builtin local pos=0
    builtin local chunk_size=4096
    while [ $pos -lt ${#data} ]; do
        builtin printf "\e_G"
        [ $pos = "0" ] && printf "a=T,f=100,"
        builtin local chunk="${data:$pos:$chunk_size}"
        pos=$(($pos+$chunk_size))
        [ $pos -lt ${#data} ] && builtin printf "m=1"
        [ ${#chunk} -gt 0 ] && builtin printf ";%s" "${chunk}"
        builtin printf "\e\\"
    done
}

transmit_png "$1"
```

Or the shell integration code (quite long so I won't paste here).

It seems unnecessary to me and make these codes much less readable.

**Comment by kovidgoyal** [comment]
Because otherwise you have no guarantee that a user hasnt aliased
a command or function in a way that breaks it.

**Reply by grondilu** [reply]
By this reasoning any bash code that doesn't use them is wrong.  This seems really over-cautious.  I know this project is not about bash scripting, but still.

**Reply by SchleimKeim** [reply]
Bash has tons of bugs as it is. Using builtin and command is more than reasonable

**Comment by kovidgoyal** [comment]
If you want robustness, use builtin and command. If you dont care,
dont. Your choice in *your* projects.

**Reply by grondilu** [reply]
Fine.  I suspected this was the reason anyway but I wasn't sure as it really seemed odd to me.  Now I'm sure.

In any case, your project, your rules indeed.  Keep up the good work!
